internals/daemon/daemon.go

```diff
@@ -766,7 +766,8 @@ func (d *Daemon) RebootIsMissing(st *state.State) error {
 // SetServiceArgs sets the provided service arguments to their respective services,
 // by passing the arguments to the service manager responsible under daemon overlord.
 func (d *Daemon) SetServiceArgs(serviceArgs map[string][]string) error {
-	return d.overlord.ServiceManager().SetServiceArgs(serviceArgs)
+	serviceArgsCreator := servstate.ServiceArgsLayerCreator(serviceArgs)
+	return d.overlord.PlanManager().AppendLayer(serviceArgsCreator)
```

Comment:
Please see the ```LayerCreator``` interface in the Plan Manager. There is a general need that ```SetServiceArgs``` as highlighted. Managers have the knowledge on how to change a part of the plan that they understand. However, these changes must happen atomically, so it must be performed while the Plan Manager maintains a lock to block updates of the plan while this mutation function is creating and applying the layer. This is what ```LayerCreator``` tries to solve. It completely compatible with the well known ```AppendLayer``` and ```CombineLayer``` methods we know well.

Reply:
Removed.